src/Sets/Zonotope.jl

```diff
@@ -371,19 +371,128 @@ nonlinear systems with uncertain parameters using conservative linearization.
+ Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.95, 0.0], [0.05 0.0; 0.0 0.1])
+ Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.05, 0.0], [0.05 0.0; 0.0 0.1])
+```
+Here, the first vector in the arguments corresponds to the zonotope's
+generator to be split, and the second vector corresponds to the exponent of
+`2^n` parts that the zonotope will be split into along the corresponding generator.
+
+Splitting of a two-dimensional zonotope along its generators:
+
+```
+julia> Z = Zonotope([1.0, 0.0], [0.1 0.0; 0.0 0.1])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.0, 0.0], [0.1 0.0; 0.0 0.1])
+
+julia> split(Z, [1, 2], [2, 2])
+16-element Array{Zonotope{Float64,Array{Float64,1},Array{Float64,2}},1}:
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.925, -0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.925, -0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.925, 0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.925, 0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.975, -0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.975, -0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.975, 0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.975, 0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.025, -0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.025, -0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.025, 0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.025, 0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.075, -0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.075, -0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.075, 0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.075, 0.075], [0.025 0.0; 0.0 0.025])
+```
+Here the zonotope is split along both of its generators, each time into four parts.
+"""
+function split(Z::Zonotope, gens::AbstractVector, n::AbstractVector)
+    @assert length(gens) == length(n) "the number of generators doesn't match the" *
+    " number of indicated partitions ($(length(gens)) and $(length(n)))"
+    @assert length(gens) <= ngens(Z) "the number of generators to split is greater" *
+    " than the number of generators of the zonotope (($(length(gens)) and $(ngens(Z)))"
+    Zs = [Z]
+    for i = 1:length(gens)
+        for j = 1:n[i]
+            km = length(Zs)
+            for k = 1:km
+                append!(Zs, split(Zs[k], gens[i]))
+            end
+            deleteat!(Zs, 1:km)
```

Comment:
I guess it is faster to collect the new zonotopes in a fresh array and replace that array in the end.

Reply:
I opened an issue to do it in the future: https://github.com/JuliaReach/LazySets.jl/issues/2224